### ISSUE TITLE
Unable to promote / Set Never work item that was manually merged into repo

### ISSUE THREAD

**Issue by SporcleHans** [forcedotcom/devops-center-feedback#252]
**Describe the bug**
Hello,

I had a work item that I accidentally manually merged via github instead through the devops center. I used the built in github tool to roll back the merge, but now the work item is stuck "in review", but it doesn't show up in the pipeline to promote. On the work item itself it just says "Merged Externally" and that I can't make any edits to the work item. It's blocking my developer from being able to sync with the next environment because devops center believes the work item is still in review. 

**To Reproduce**
1. Manually merge a pull request from a work item
2. roll back the pull request

**Expected behavior**
Being able to promote the work item, or set it to "never"

**Comment by franciscoperezsammartino** [comment]
Hi SporcleHans, DevOps Center does not support RollBack an external merge. Still it should show the Work Item in the Approved column to be deployed (because it was already merged) into the next stage.

My suggestion would be abandon that Work Item and create a new one to keep on your development. As you said that you rollback the changes, the pipeline stage should not have any changes from the merged work item.

**Comment by SporcleHans** [comment]
Thanks that is what we did, but unfortunately the main issue is that it will not let us sync the sandbox with the next development stage as it says that the github merged work item is still "In Progress"

**Comment by franciscoperezsammartino** [comment]
I can provide a work around you can try to get out of this situation, but it is a bit involved. 
1. We need to disable 2  validation rules from Work_Item__c.
2.  Got to Setup -> Object Manager -> Work_Item__c -> Validation rules
3. You should see among the validation rules, two of them called **PromotedFromEnvironment_Changed** and **PromotedFromEnvironment_Set**
4. Edit both of them and Set the Active boolean to be false.
Without this validation rules now we can make a change to the work item. 
5. Go to LEX and get into the work item you are having problems layout (not in DevOps Center, but in LEX)
6. In the Work_Item__c there is a field called Promoted_From_Environment__c
7. Edit this field and set its value to be empty.
8. Save the changes


After this please re-activate the validation rules, and retry the sync you were trying to make.

**Comment by sdavis-AWP** [comment]
^^ This. If a WI fails to deploy in a downstream environment after initial deployment from dev environment, there's no way to roll back the work item, change to never, or even delete it(without manually deleting the records). It stays stuck in the pipeline. Then if you create a new work item with the same metadata/source members, it will be blocked from deployment due to dependency on the failed work item.

**Comment by MilaMerkle** [comment]
Hi All, 
just yesterday, I faced similar issue. I merged WorkItem in GitHub succesfully but received an error when I tried to promote it via DevOps to complete the merge. Therefore, that workitem got stuck because I was not able to promote it due to error and I was not able to change status to Never because it has been merged outside of DevOps. 

What worked for me was to: 

- Revert changes it that branch in GitHub
- Delete workitem with related remote change and change submission records via Salesforce Inspector. 

Not sure if this is the correct way but it worked for me. My pipeline got unblocked and I can work with it once again. 

The drawback of this workaround is that DevOps no longer recognise your changes and you have to delete in SF what you did and do it all over again. 

Hope it helps. 
Mila

**Comment by jhaltermann** [comment]
I had the same issue and for me it worked to: 

- Revert the merge-commit in github

- Use Salesforce inspector to set work item to never `sf_devops__Concluded__c = NEVERED`

I could then create a new WI and add the changes again
